pd/src/components/staking.rs

```diff
@@ -634,11 +614,51 @@ impl Component for Staking {
             }
         }
 
-        // The validator definitions have been completely verified, so we can add them to the pending list:
-        let mut definitions = tx.validator_definitions().map(|v| v.to_owned()).collect();
-        self.new_validator_definitions
-            .definitions
-            .append(&mut definitions);
+        // The validator definitions have been completely verified, so we can add them to the JMT
+        let definitions = tx.validator_definitions().map(|v| v.to_owned());
+        let cur_epoch = self.overlay.get_current_epoch().await?;
+
+        for v in definitions {
+            let existing_v = self.overlay.validator(&v.validator.identity_key).await?;
+
+            if let Some(_existing_v) = existing_v {
```

Comment:
tioli: I think this could be combined into
```rust
if self.overlay.validator(&v.validator.identity_key).await?.is_some() {
    // This is an existing validator....
} else {
    // This is a new validator ...
}
```